### DISCUSSION TITLE
Can Github PAT be deleted after bootstrap?

### DISCUSSION THREAD

**Discussion by pozniakas** [fluxcd/flux2#4328] [Q&A] [answered]
Hi, I see that after the bootstrap of FluxCD into my cluster Deploy key is generated. Does this mean that the PAT used for bootstrap can be deleted, because rest of the actions will be used by Deploy key?

**Comment by stefanprodan** [comment, marked as answer]
> Note that when the PAT is removed or when it expires, the GitHub deploy key will stop working. To regenerate the deploy key, delete the flux-system secret from the cluster and re-run the bootstrap command using a valid GitHub PAT.

From doc https://fluxcd.io/flux/installation/bootstrap/github/#github-deploy-keys

**Reply by pozniakas** [reply]
Somehow I missed this. Thank you so much! ❤️

**Reply by stefanprodan** [reply]
If you don't want to link the deploy key to your PAT, see this doc section on how to use a standalone SSH Key: https://fluxcd.io/flux/installation/bootstrap/github/#bootstrap-without-a-github-pat

**Reply by 3213-cmd** [reply]
~~Hi, do you know if this Is this still the case? I just bootstrapped using a PAT, which created a deploy key. After revoking/deleting the PAT. The deploy kill still seems to be working. Thank you for your work!~~

EDIT:
In case somebody stumbles upon this and wonders if that is actually the case.

It seems like it may take some time for the key to be "garbage collected".

Based on the [offical Github Documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#deleting-a-personal-access-token): _If you delete a personal access token that was used to create a deploy key, the deploy key will also be deleted._

and some [GitHub Discussions](https://github.com/orgs/community/discussions/22513#discussioncomment-10355601) I found.
